dask/tests/test_distributed.py

```diff
@@ -744,3 +744,20 @@ def __init__(self, val):
     )
 
     assert (await c.compute(result)).val == 170
+
+
+def test_set_index_no_resursion_error(c):
+    # see: https://github.com/dask/dask/issues/8955
+    pytest.importorskip("numpy")
+    pytest.importorskip("pandas")
+    pytest.importorskip("dask.dataframe")
+    try:
+        ddf = (
+            dask.datasets.timeseries(start="2000-01-01", end="2000-07-01", freq="12h")
+            .reset_index()
+            .astype({"timestamp": str})
+        )
+        ddf = ddf.set_index("timestamp", sorted=True)
+        ddf.compute()
+    except RecursionError as re:
+        raise (re)
```

Comment:
This seems like a no-op to me.  Maybe drop the entire try-except?

Reply:
Yeah -- I was trying to convey that we're testing to confirm that we've corrected the `RecursionError` issue.

Perhaps explicitly calling `pytest.fail` here would make more sense. If not, dropping `try/except` and adding more comments would make sense.